maven-plugin/src/main/java/org/wildfly/galleon/maven/ArtifactListBuilder.java

```diff
@@ -114,6 +127,24 @@ private static Model readModel(final Path pomXml) throws IOException {
         }
     }
 
+    private static Charset getEncoding(Path pomXml) throws IOException {
+        Charset charset = StandardCharsets.UTF_8;
+        try (FileReader fileReader = new FileReader(pomXml.toFile())) {
+            XMLStreamReader xmlReader = XML_INPUT_FACTORY.createXMLStreamReader(fileReader);
+            try {
+                String encoding = xmlReader.getCharacterEncodingScheme();
+                if (encoding != null) {
+                    charset = Charset.forName(encoding);
+                }
+            } finally {
+                xmlReader.close();
+            }
+        } catch (XMLStreamException ex) {
+            throw new IOException("Exception " + ex.getMessage() + " when retrieving encoding for " + pomXml, ex);
```

Comment:
Let's not duplicate error messages. "Failed to retrieve encoding for xxx" will be good enough.